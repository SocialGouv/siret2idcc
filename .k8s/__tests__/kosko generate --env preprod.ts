//

import { getEnvManifests } from "@socialgouv/kosko-charts/testing";
import { project } from "@socialgouv/kosko-charts/testing/fake/gitlab-ci.env";

jest.setTimeout(1000 * 60);
test("kosko generate --preprod", async () => {
  expect(
    await getEnvManifests("preprod", "", {
      ...project("siret2idcc").preprod,
      KUBE_NAMESPACE: "siret2idcc-65-preprod-dev2",
      RANCHER_PROJECT_ID: "c-bd7z2:p-dtlsm",
    })
  ).toMatchSnapshot();
});
