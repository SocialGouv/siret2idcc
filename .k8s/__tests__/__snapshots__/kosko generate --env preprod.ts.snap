// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kosko generate --preprod 1`] = `
"---
metadata:
  annotations:
    field.cattle.io/creatorId: gitlab
    field.cattle.io/projectId: 'c-kk8xm:p-2h6lh'
    git/branch: v1.2.3
    git/remote: >-
      https://gitlab-ci-token:[MASKED]@gitlab.factory.social.gouv.fr/SocialGouv/siret2idcc.git
    app.gitlab.com/app: socialgouv-siret2idcc
    app.gitlab.com/env: preprod-dev2
    app.gitlab.com/env.name: preprod-dev2
  labels:
    azure-pg-admin-user: siret2idcc
    application: v1-2-3-siret2idcc
    owner: siret2idcc
    team: siret2idcc
    cert: wildcard
  name: siret2idcc-85-preprod-dev2
apiVersion: v1
kind: Namespace
---
metadata:
  annotations:
    app.gitlab.com/app: socialgouv-siret2idcc
    app.gitlab.com/env: preprod-dev2
    app.gitlab.com/env.name: preprod-dev2
  labels:
    app: app
    application: v1-2-3-siret2idcc
    owner: siret2idcc
    team: siret2idcc
    cert: wildcard
  name: app
  namespace: siret2idcc-85-preprod-dev2
spec:
  replicas: 1
  selector:
    matchLabels:
      app: app
  template:
    metadata:
      annotations:
        app.gitlab.com/app: socialgouv-siret2idcc
        app.gitlab.com/env: preprod-dev2
        app.gitlab.com/env.name: preprod-dev2
      labels:
        app: app
        application: v1-2-3-siret2idcc
        owner: siret2idcc
        team: siret2idcc
        cert: wildcard
    spec:
      containers:
        - image: >-
            registry.gitlab.factory.social.gouv.fr/socialgouv/siret2idcc/app:1.2.3
          livenessProbe:
            failureThreshold: 6
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 30
            periodSeconds: 5
            timeoutSeconds: 5
          name: app
          ports:
            - containerPort: 3000
              name: http
          readinessProbe:
            failureThreshold: 15
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 0
            periodSeconds: 5
            successThreshold: 1
            timeoutSeconds: 1
          resources:
            limits:
              cpu: 500m
              memory: 1280Mi
            requests:
              cpu: 100m
              memory: 128Mi
          startupProbe:
            failureThreshold: 12
            httpGet:
              path: /healthz
              port: http
            periodSeconds: 5
apiVersion: apps/v1
kind: Deployment
---
metadata:
  labels:
    app: app
    application: v1-2-3-siret2idcc
    owner: siret2idcc
    team: siret2idcc
    cert: wildcard
  name: app
  annotations:
    app.gitlab.com/app: socialgouv-siret2idcc
    app.gitlab.com/env: preprod-dev2
    app.gitlab.com/env.name: preprod-dev2
  namespace: siret2idcc-85-preprod-dev2
spec:
  ports:
    - name: http
      port: 80
      targetPort: 3000
  selector:
    app: app
  type: ClusterIP
apiVersion: v1
kind: Service
---
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    app.gitlab.com/app: socialgouv-siret2idcc
    app.gitlab.com/env: preprod-dev2
    app.gitlab.com/env.name: preprod-dev2
  labels:
    app: app
    application: v1-2-3-siret2idcc
    owner: siret2idcc
    team: siret2idcc
    cert: wildcard
  name: app
  namespace: siret2idcc-85-preprod-dev2
spec:
  rules:
    - host: preprod-siret2idcc.dev2.fabrique.social.gouv.fr
      http:
        paths:
          - backend:
              serviceName: app
              servicePort: 80
            path: /
  tls:
    - hosts:
        - preprod-siret2idcc.dev2.fabrique.social.gouv.fr
      secretName: wildcard-crt
apiVersion: extensions/v1beta1
kind: Ingress
"
`;
